dali/python/nvidia/dali/eager.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+
+import nvidia.dali.backend as _b
+import nvidia.dali.internal as _internal
+import nvidia.dali.ops as _ops
+import nvidia.dali.tensors as _tensors
+from nvidia.dali._utils.eager_util import _Classification, _transform_data_to_tensorlist
+
+
+_statefull_operators = {
+    'decoders__ImageRandomCrop',
+    'noise__Gaussian',
+    'noise__SaltAndPepper',
+    'noise__Shot',
+    'segmentation__RandomMaskPixel',
+    'segmentation__RandomObjectBBox',
+    'FastResizeCropMirror',
+    'Jitter',
+    'ROIRandomCrop',
+    'RandomBBoxCrop',
+    'RandomResizedCrop',
+    'ResizeCropMirror',
+}
+
+
+_generator_operators = {
+    'readers__COCO',
+    'readers__Caffe',
+    'readers__Caffe2',
+    'readers__File',
+    'readers__MXNet',
+    'readers__NemoAsr',
+    'readers__Numpy',
+    'readers__Sequence',
+    'readers__TFRecord',
+    'readers__Video',
+    'readers__VideoResize',
+    'readers__Webdataset',
+    'random__CoinFlip',
+    'random__Normal',
+    'random__Uniform',
+    'BatchPermutation',
+}
+
+
+_stateless_operators_cache = {}
+
+
+def _eager_op_base_factory(op_class, op_name, num_inputs, call_args_names):
+    class EagerOperatorBase(op_class):
```

Comment:
I guess there is no escape from op class.

Reply:
Just reusing the code. The class has to be somewhere because of the backend implementation.